docs/snakefiles/rules.rst

```diff
@@ -586,7 +585,7 @@ It can for example be used to extract a value from inside an input file.
         output:
             "samples.id",
         params:
-            id=parse_input(input.samples, parser=extract_id)
+            id=lambda w, input: parse_input(input.samples, parser=extract_id)
```

Comment:
The point of the helper and in particular the input.xyz helpers is to not having to write lambda expressions here. Why would it be needed now?

Reply:
The tests have a wildcard to specify the filename, and I can only get it to work with a lambda function.

Reply:
I am getting an error where `infile` is a `<function RuleItemProxy._deferred_get.<locals>.inner at 0x7f2f2ac5a8e0>` and python throws:
```
>       with open(infile, "r") as fh:
E       TypeError: expected str, bytes or os.PathLike object, not function
```

Reply:
Then we have a bug there. It should work with the syntax that we had in the test.

Reply:
I have added a commit to fix the issue. In the testcase, it would be good to test that as well. Currently, the testcase requires a more complex lambda because the file argument depends on wildcards, which is probably a special case.